modules/openshift-developer-cli-openshift-cluster-objects.adoc

```diff
@@ -0,0 +1,87 @@
+
+The Pod created by {odo-title} executes two init containers:
+. The copy-supervisord init container.
+. The copy-files-to-volume init container.
+
+=== `copy-supervisord`
+
+The copy-supervisord init container copies necessary files onto an emptyDir Volume, the main application container utilizes these files from the emptyDir Volume.
+
+.The following files are copied onto the emptyDir Volume:
+* Binaries:
+** `go-init` is a minimal init system. It runs as the first process (PID 1) inside the application container. go-init starts the SupervisorD daemon which runs the developer code. go-init is required to handle orphaned processes.
+** `SupervisorD` is a process control system. It watches over configured processes and ensures that they are running. It also restarts services when necessary. For {odo-title}, SupervisorD executes and monitors the developer code.
+
+* Configuration files:
+** `supervisor.conf` is the configuration file necessary for the SupervisorD daemon to start.
+* Scripts:
+** `assemble-and-restart` is an OpenShift S2I concept to build and/or deploy user-source code. The assemble-and-restart script first assembles the user source code inside the application container and then restarts `SupervisorD` for user changes to take effect.
+** `Run` is an OpenShift S2I concept of executing the assembled source code. The run script executes the assembled code created by the `assemble-and-restart` script.
+** `s2i-setup` is a script that creates files and directories which are necessary for the `assemble-and-restart` and run scripts to execute successfully. The script is executed whenever the application container starts.
+
+* Directories:
+** `language-scripts`: Openshift S2I allows custom `assemble` and `run` scripts. A few language specific custom scripts are present in the `language-scripts` directory. The custom scripts provide additional configuration to make {odo-title} debug work.
+
+The `emtpyDir Volume` is mounted at `/opt/odo` mount point for both the init container and the application container.
+
+=== `copy-files-to-volume`
+The `copy-files-to-volume` init container copies files that are in `/opt/app-root` in S2I builder image onto the Persistent Volume. The volume is then mounted at the same location (`/opt/app-root`) in an application container.  
+
+Without the `PersistentVolume` on `/opt/app-root` the data in this directory is lost when `PersistentVolumeClaim` is mounted at the same location.
+
+The `PVC` is mounted at `/mnt` mount point inside the init container.
+
+== Application container
+Application container is the main container inside of which the user-source code executes. 
+
+Application container is mounted with two Volumes:
+* emptyDir Volume mounted at /opt/odo
+* The Persistent Volume mounted at /opt/app-root
+
+The `go-init` is executed as the first process inside the application container. The `go-init` process then starts the `SupervisorD` daemon.
```

Comment:
In this and the next sentence, `go-init` and `SupervisorD` should only have "The" in front of them if you have "process" after it.